cmd/cmd.go

```diff
@@ -277,12 +280,65 @@ func createBlob(cmd *cobra.Command, client *api.Client, path string) (string, er
 	}
 
 	digest := fmt.Sprintf("sha256:%x", hash.Sum(nil))
-	if err = client.CreateBlob(cmd.Context(), digest, bin); err != nil {
+
+	// Use our new CreateBlob request which will include the file path
+	// The server checks for that file and if the server is local, it will copy the file over
+	// If the local copy fails, the server will continue to the default local copy
+	// If that fails, it will continue with the server POST
+	err = CreateBlob(cmd.Context(), path, digest, bin)
+	if errors.Is(err, ErrBlobExists) {
+		return digest, nil
+	}
+
+	if err != nil {
 		return "", err
 	}
+
 	return digest, nil
 }
 
+func CreateBlob(ctx context.Context, src, digest string, r *os.File) (error) {
```

Comment:
Is a separate function required here? If so, I don't think we need to expose it (by making it uppercase)